drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -617,6 +618,20 @@ static int pmw3360_async_init_fw_load_verify(struct pmw3360_data *dev_data)
 
 static void irq_handler(struct device *gpiob, struct gpio_callback *cb,
 			u32_t pins)
+{
+	int err;
+
+	err = gpio_pin_disable_callback(pmw3360_data.irq_gpio_dev,
+					PMW3360_IRQ_GPIO_PIN);
```

Comment:
The callback is disabled here but not re-enabled after the handler is executed.
But is this disabling actually needed? It is okay to submit a work item again while the previously submitted one is still pending in the queue.

Reply:
ISR does not clear the status of the sensor so IRQ line will always be active. If we don't disable the callbacks CPU will never get out from this ISR. Other sensors disable callback in ISR the same way (probably for the same reason).

However it seems that I indeed should re-enable this callback after calling the handler - at least this is what other sensors do. I cannot however fix it now as this requires a change in our app. Let me create another PR for it - I will fix both sensors and app there.

Re-submission of work automatically cancels the previous submission. We cannot submit new work before the previous is handled - otherwise we would miss the trigger.

Reply:
I checked the application and we have a code in place to disable the callback so the bug was only in the PAW3212. I have added a fix in a separate commit.

Reply:
Fixed